### PR TITLE
Add recommender module

### PR DESCRIPTION
**Pull request by NoB0**
Creates a recommender module with an abstract class and a subclass for slot-based recommendation.

Upon approval, the recommendation process in `DialogueManager` should be refactored and tests should be added (issue #145).

Part of #49

### CODE REVIEW

File: moviebot/recommender/slot_based_recommender_model.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from typing import Any, Dict, List
+
+from moviebot.database.database import DataBase
+from moviebot.dialogue_manager.dialogue_state import DialogueState
+from moviebot.ontology.ontology import Ontology
+from moviebot.recommender.recommender_model import RecommenderModel
+
+
+class SlotBasedRecommenderModel(RecommenderModel):
+    def __init__(self, db: DataBase, ontology: Ontology) -> None:
```
Comment (WerLaj) [inline moviebot/recommender/slot_based_recommender_model.py:12]: Do we actually need here __init__ method? It is the same as in `RecommenderModel` class.

**Review by WerLaj — APPROVED** [review]
LGTM

File: moviebot/recommender/slot_based_recommender_model.py (hunk 1 above)
Comment (NoB0) [reply moviebot/recommender/slot_based_recommender_model.py]: Good point! I removed it.

File: moviebot/recommender/recommender_model.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        Returns:
+            Recommended movies.
+
+        Raises:
+            NotImplementedError: If not implemented in derived class.
+        """
+        raise NotImplementedError
+
+    def previous_recommend_items(self) -> List[Dict[str, Any]]:
+        """Returns the list of previously recommended movies."""
```
[the remark this answers is not in the document]
Comment (NoB0) [reply moviebot/recommender/recommender_model.py]: Done.

**Review by vinaysetty — APPROVED** [review]
LGTM.

**Pull request merged**